packages/desktop-client/src/components/util/PercentInput.tsx

```diff
@@ -0,0 +1,125 @@
+import React, {
+  type Ref,
+  useRef,
+  useState,
+  useEffect,
+  type FocusEventHandler,
+  type FocusEvent,
+} from 'react';
+
+import { evalArithmetic } from 'loot-core/src/shared/arithmetic';
+
+import { useMergedRefs } from '../../hooks/useMergedRefs';
+import { type CSSProperties } from '../../style';
+import { Input } from '../common/Input';
+import { useFormat } from '../spreadsheet/useFormat';
+
+type PercentInputProps = {
+  id?: string;
+  inputRef?: Ref<HTMLInputElement>;
+  value: number;
```

Comment:
Isn't this optional? Below it sets a default of `= 0`.
```suggestion
  value?: number;
```